libcalico-go/lib/backend/k8s/k8s.go

```diff
@@ -217,6 +217,12 @@ func NewKubeClient(ca *apiconfig.CalicoAPIConfigSpec) (api.Client, error) {
 		libapiv3.KindIPAMConfig,
 		resources.NewIPAMConfigClient(cs, crdClientV1),
 	)
+	kubeClient.registerResourceClient(
```

Comment:
I am realizing now that this would have been a bit simpler if we simply had two different backend clients - e.g.,

```
registerResourceClient( NewV3BlockAffinityClient )
registerResourceClient (NewV1BlockAffinityClient )
```

But that's a project for another day.